src/client.js

```diff
@@ -155,22 +155,19 @@ exports.createClient = function(reference, name) {
 
   // Client class constructor
   var Client = function(options) {
-    if (options && options.baseUrl && options.rootUrl) {
-      throw new Error('baseUrl and rootUrl are mutually exlcusive. Prefer rootUrl.');
+    if (options && options.baseUrl) {
+      throw new Error('baseUrl has been deprecated!');
     }
     this._options = _.defaults({}, options || {}, {
-      baseUrl:          reference.baseUrl        || '',
       exchangePrefix:   reference.exchangePrefix || '',
+      // We can remove the second half of this assignment once all api definitions are upgraded to have `name`
+      serviceName: reference.name || reference.baseUrl.split('//')[1].split('.')[0],
+      serviceVersion: 'v' + (reference.version + 1),
     }, _defaultOptions);
 
-    // Remove possible trailing slash from baseUrl
-    this._options.baseUrl = this._options.baseUrl.replace(/\/$/, '');
+    assert(this._options.rootUrl, 'Must provide a rootUrl or set the env var TASKCLUSTER_ROOT_URL');
```

Comment:
Maybe worth having a `tcUrl.isValidRootUrl(..)` to check this, and clean its trailing slash..

Reply:
The only problem is that I think we're pretty liberal with what a valid root url is, so I'm not sure how much we could validate really.

Reply:
Well we can be as conservative as we want :)

I was thinking
 * has schema and it's http or https
 * has a hostname
 * has no path, or has path / (we should enforce one of those two)